File: 백준/Bronze/15751. Teleportation/Teleportation.py
import sys
input=lambda:sys.stdin.readline().rstrip()
a,b,x,y=map(int,input().split())
a,b=min(a,b),max(a,b)
x,y=min(x,y),max(x,y)
print(min(b-a,abs(x-a)+abs(y-b)))

File: 백준/Bronze/15751. Teleportation/Teleportation.cc
#include <iostream>
using namespace std;
int main(void) {
    int a, b, x, y, t1, rlt;
    cin >> a >> b >> x >> y;
    if (a > b) {
        t1 = a;
        a = b;
        b = t1;
    }
    if (x > y) {
        t1 = x;
        x = y;
        y = t1;
    }
    rlt = min(b - a, abs(x - a) + abs(y - b));
    cout << rlt;
    return 0;
}